app/javascript/components/admin/roles/forms/EditRoleForm.jsx

```diff
@@ -16,30 +18,36 @@ import RolePermissionRow from '../RolePermissionRow';
 import { useAuth } from '../../../../contexts/auth/AuthProvider';
 
 export default function EditRoleForm({ role }) {
-  const methods = useForm(editRoleFormConfig);
+  const methodsRoleName = useForm(editRoleFormConfigRoleName);
+  const methodsRoomLimit = useForm(editRoleFormConfigRoomLimit);
   const updateRoleAPI = useUpdateRole(role.id);
   const updateRolePermission = () => useUpdateRolePermission();
-  const { defaultValues } = editRoleFormConfig;
-  const fields = editRoleFormFields;
-  fields.name.placeHolder = defaultValues.name;
+  const updateAPI = updateRolePermission();
+  const fieldsRoleName = editRoleFormFieldsRoleName;
+  const fieldsRoomLimit = editRoleFormFieldsRoomLimit;
   const roomConfigs = useRoomConfigs();
   const { data: rolePermissions, isLoading: rolePermissionsIsLoading } = useRolePermissions(role.id);
   const currentUser = useAuth();
 
   useEffect(
     () => {
-      methods.setValue('name', role.name);
+      methodsRoleName.setValue('name', role.name);
     },
     [role.name],
   );
 
   if (roomConfigs.isLoading || rolePermissionsIsLoading) return <Spinner />;
+  fieldsRoomLimit.roomLimit.placeHolder = rolePermissions.RoomLimit;
 
   return (
     <div>
       <Stack>
-        <Form methods={methods} onBlur={(e) => updateRoleAPI.mutate({ name: e.target.value })}>
-          <FormControl field={fields.name} type="text" />
+        <Form methods={methodsRoleName} onBlur={(e) => updateRoleAPI.mutate({ name: e.target.value })}>
```

Comment:
I suspect that `react-hook-form` here isn't handling neither the submission nor the form data.
We're bypassing it,
A great example of the implication of that is that client side validations have no effect besides providing a feedback (it will not prevent the API call to be made).
Another thing that may cause an issue is the lack of preventing of the browser defaults for the `onblur`  (haven't checked the default behavior and its implication) on the form.

Reply:
After ensuring the good use of react hook form I recommend that you (if possible) separate both of these forms into their separate components and make this a layout component that renders both of these forms instead of this `EditRoleForm` that has two forms,

Reply:
Yes you are right, I was having issues with this. Honestly I find the best solution to just create an update button beside each of the inputs. I feel this would also align with the rest of the application